auth/src/main/java/org/aerogear/mobile/auth/authenticator/AbstractAuthenticator.java

```diff
@@ -36,8 +36,9 @@ public AbstractAuthenticator(@NonNull final ServiceConfiguration serviceConfig)
      * Logout the given principal
      *
      * @param principal principal to be log out
+     * @param callback the callback function to be invoked
      */
-    public abstract void logout(final UserPrincipal principal);
+    public abstract void logout(final UserPrincipal principal, final Callback<UserPrincipal> callback);
```

Comment:
I would prefer of instead of a call back you use a promise style API.  Callbacks are often written as anonymous inner classes in Activities which can leak references and cause crashes.

Ultimately we will probably want to have our exposed APIs wrap something like RxJava behind the scenes

Reply:
@secondsun I would agree, but I think it should be consistent across all the modules. Perhaps something should be provided by the core module?

For now, let's use callback for this one as login method is already using callbacks.

Reply:
@secondsun Sounds like a plan. Do you mind opening an issue in this repo to discuss a bit more widely? For this PR I am following the current Callback style that is used currently by other functions in the SDK.  I think it would be good to have some alignment of the best approach for this as it will affect the example/mobile-security apps too with how they do callbacks.

Reply:
If you missed the issue further down : https://github.com/aerogear/aerogear-android-sdk/issues/122